### PR TITLE
Support render props alternative due to deprecated findDOMNode()

### PR DESCRIPTION
**Pull request by atomiks**
Continued from discussion in #27 

Until React supports refs on `Fragments` this seems to be required for strict mode. Anyone else want to chime in?

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -5,7 +5,9 @@ type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
 
 export interface TippyProps extends Omit<Props, 'content'> {
   content: React.ReactElement<any> | string
-  children: React.ReactElement<any>
+  children: (
+    ref: React.RefObject
+  ) => React.ReactElement<any> | React.ReactElement<any>
```
Comment (MikeDombo) [inline index.d.ts]: Think you need some more parentheses.

```
children: ((
    ref: React.RefObject
  ) => React.ReactElement<any>) | React.ReactElement<any>
```

Comment (atomiks) [reply index.d.ts]: Prettier seems to change that to:

```ts
children:
  | (ref: React.RefObject) => React.ReactElement<any>
  | React.ReactElement<any>
```

Is that valid?

**Comment by atomiks** [comment]
Yes, without `findDOMNode()` we need to do:

```jsx
<Tippy>
  {ref => <button ref={ref}>text</button>}
</Tippy>
```

Instead of just:

```jsx
<Tippy>
  <button>text</button>
</Tippy>
```

Unless `<Fragment>` can have a `ref`. 

This seems to be a use case the React team didn't consider, should we make an issue on the React repo making them aware of this? The deprecation of `findDOMNode()` requires the usage of render props which is annoying to write without any benefits, since the user doesn't care about exposing the outermost DOM node to `<Tippy>`, they're doing it on purpose

**Comment by atomiks** [comment]
There is now a `Fragment` ref RFC just posted... I'm going to hold off on this as I don't like how cumbersome it is. I will switch to the `Fragment` ref version when/if it gets merged in some new version. For now you'll have to live with the warning in strict mode... 😭

**Comment by atomiks** [comment]
v2 won't have this problem

**Pull request closed without merging**